backend/app/readers/readers_instantiator.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from app.models.settings.settings import settings
+from app.readers.reader.reader import Reader
+from app.readers.reader.wikipedia_reader import WikipediaReader
+
+
+class ReadersInstantiator:
+    """
+    An Instantiator for the different Readers.
+    """
+
+    def __init__(self) -> None:
+        self.__record_types: frozenset[str] | None = self.__get_record_types()
+        self.__file_output_paths: list[Path] | None = self.__get_output_paths()
+
+    def __get_record_types(self) -> frozenset[str] | None:
+        """Retrieve a frozenset of record types from settings."""
+
+        if settings.record_types:
+            return settings.record_types
+        return None
+
+    def __get_output_paths(self) -> list[Path] | None:
+        """Retrieve output file names from settings and return a list of file output paths."""
+
+        if settings.output_file_names:
+            return [
+                Path(__file__).parent.parent / "data" / file_name
+                for file_name in settings.output_file_names
+            ]
+
+        return None
+
+    def provide_readers(self) -> tuple[Reader, ...] | None:
+        """Return a tuple of Readers depending on the record type obtained from settings.
+        Readers are instantiated with Paths that contain the corresponding record types in the Path names.
+        """
+        if self.__record_types and self.__file_output_paths:
```

Comment:
I can see why you'd have this class separate from AllSources for testing purposes. But you can just make it a freestanding function that takes `settings`, no need to have a class.

```
def create_readers(settings) -> tuple[Reader, ...]:
```

Then you can test the factory function above separately from `AllSourceReader`.